src/main/java/com/microsoft/jenkins/azuread/AzureSecurityRealm.java

```diff
@@ -641,6 +651,14 @@ public SecurityComponents createSecurityComponents() {
         });
     }
 
+    private static @NonNull User getByIdOrCreate(AzureAdUser user) {
```

Comment:
Ideally core would handle this with a separate method that is annotated to not return null as it can't return null with create true